src/message-authentication/hash-based.md

```diff
@@ -0,0 +1,63 @@
+# Hash-Based Message Authentication (HMAC)
+
+HMACs are [MACs](index.html) based on a hash algorithm and are named after that algorithm; e.g.
+`HMAC-SHA256`. Usually, a HMAC algorithm will concatenate the data and the secret then hash them
+together to create the signature.
+
+The [`hmac`] crate provides a wrapper type [`Hmac`] that implements the trait [`Mac`] for any type
+that implements [`Digest`]: for example, `sha2::Sha256`. See
+[Data Hashing](../hashing-data/index.html) for information on hash algorithms.
+
+A good example of HMACs in use are with JWTs (JSON web tokens). Data that is roughly
+`{"username": "ferris", "issued": "2023-10-02T21:00:00"}` is signed with the server's secret, then
+given to a client upon successful login. The client can then provide this key back to the server to
```

Comment:
"key" --> "token" ?

Reply:
Yeah, thanks. I rewrote this section if you could take another look.